src/openrct2/config/IniReader.cpp

```diff
@@ -0,0 +1,384 @@
+ * For more information, visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#include <initializer_list>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+#include "../common.h"
+#include "../core/FileStream.hpp"
+#include "../core/String.hpp"
+#include "../core/StringBuilder.hpp"
+#include "IniReader.hpp"
+
+/**
+ * Simple tuple (start, length) representing a text span in a buffer.
+ */
+struct Span
+{
+    size_t Start    = 0;
+    size_t Length   = 0;
+
+    Span() = default;
+    Span(size_t start, size_t length)
+        : Start(start),
+          Length(length)
+    {
+    }
+};
+
+/**
+ * Simple tuple (start, end) inclusive representing a range of lines.
+ */
+struct LineRange
+{
+    size_t Start    = 0;
+    size_t End      = 0;
+
+    LineRange() = default;
+    LineRange(size_t start, size_t end)
+        : Start(start),
+          End(end)
+    {
+    }
+};
+
+class IniReader final : public IIniReader
+{
+private:
+    std::vector<uint8>                              _buffer;
+    std::vector<Span>                               _lines;
+    std::unordered_map<std::string, LineRange>      _sections;
+    std::unordered_map<std::string, std::string>    _values;
+
+public:
+    IniReader(IStream * stream)
+    {
+        uint64 length = stream->GetLength() - stream->GetPosition();
+        _buffer.resize(length);
+        stream->Read(_buffer.data(), length);
+
+        RemoveBOM();
+
+        // Ensure there is a null terminator on the end, this is
+        // mainly for ParseLines's sake
+        if (_buffer.empty() || _buffer[length - 1] != 0)
+        {
+            _buffer.push_back(0);
+        }
+
+        ParseLines();
+        ParseSections();
+    }
+
+    bool ReadSection(const std::string &name) override
+    {
+        auto it = _sections.find(name);
+        if (it == _sections.end())
+        {
+            return false;
+        }
+
+        ParseSectionValues(it->second);
+        return true;
+    }
+
+    bool GetBoolean(const std::string &name, bool defaultValue) const override
+    {
+        bool result = defaultValue;
+        std::string value;
+        if (TryGetString(name, &value))
+        {
+            result = String::Equals(value, "true", true);
+        }
+        return result;
+    }
+
+    sint32 GetSint32(const std::string &name, sint32 defaultValue) const override
+    {
+        sint32 result = defaultValue;
+        std::string value;
+        if (TryGetString(name, &value))
+        {
+            result = std::stoi(value);
```

Comment:
Just so you know: https://github.com/OpenRCT2/OpenRCT2/pull/5240/commits/8143be0707b9252652dc9a99a28c1927054a856b#diff-e872ff6bb0551fec33f38f976677859fR48